lib/benchpark/experiment.py

```diff
@@ -70,12 +70,43 @@ def compute_modifiers_section(self):
 
     def compute_applications_section(self):
         # TODO: is there some reasonable default?
+        variables = {}
+        variables["n_ranks"] = num_procs  # TODO: num_procs will be defined in the child...
+        n_resources = num_procs
+
         raise NotImplementedError(
             "Each experiment must implement compute_applications_section"
         )
 
     def compute_spack_section(self):
         # TODO: is there some reasonable default based on known variable names?
+        system_specs = {}
+        system_specs["compiler"] = "default-compiler"
+        system_specs["mpi"] = "default-mpi"
+        system_specs["lapack"] = "lapack"   #TODO: can we define this in lapack experiment?
+
+        package_specs = {}
+        package_specs[system_specs["mpi"]] = (
+            {}
+        )  # empty package_specs value implies external package
+
+        # TODO: Can we define this in lapack/blas experiment?
+        package_specs[system_specs["lapack"]] = (
+            {}
+        )  # empty package_specs value implies external package
+
+        # TODO: is there a way to define hypre experiment, where this would live?
+        package_specs["hypre"] = {
+            "pkg_spec": f"hypre@{hypre_version} +mpi+mixedint~fortran",
+            "compiler": system_specs["compiler"],
+        }
+
+        # TODO: is there a way to generically do this?
+        package_specs[app_name] = {
+            "pkg_spec": f"{app_name}@{app_version} +mpi",
+            "compiler": system_specs["compiler"],
+        }
+
         raise NotImplementedError(
             "Each experiment must implement compute_spack_section"
         )
```

Comment:
If this does anything, it can't raise `NotImplementedError`. FWIW `benchpark audit` currently makes sure that the lowest-level definition (e.g. what lives in `exp_repo`) defines this method.